[material-ui][Button] Add loading feature to Button (#42987)
mui/material-ui: docs/data/material/components/buttons/buttons.md

{{"demo": "IconButtonColors.js"}}

> ### Loading with badge

Reviewer: I think it would be better to separate loading and badge demos. Is there a reason to mix both?

Reviewer 2: Done. The reason was to ensure a case like https://github.com/mui/material-ui/pull/43436, but it's better to separate.

{{"demo": "IconButtonColors.js"}}

### Loading

> Use `loading` prop to set icon buttons in a loading state and disable interactions.

Reviewer: `loading` is universally accepted, so there isn't much to discuss IMO.

> Use `loading` prop to set icon buttons in a loading state and disable interactions.

Reviewer: Nah, `loading` is universally established.

> Use `loading` prop to set icon buttons in a loading state and disable interactions.

Reviewer 3: Alright ok

{{"demo": "InputFileUpload.js"}}

> ### Loading button

Reviewer: This should be an h2, otherwise it shows as a subheading of "File upload"

```suggestion
## Loading button
```

> Use the `loading` prop to give buttons a loading state and disable interactions

Reviewer: ```suggestion
Use the `loading` prop to set buttons in a loading state and disable interactions.
```

{{"demo": "InputFileUpload.js"}}

> ## Loading button

Reviewer: ```suggestion
## Loading
```
